### ISSUE TITLE
Doc request - is nightmare.js headless?

### ISSUE THREAD

**Issue by RichardJECooke** [segmentio/nightmare#744]
Hi, my company is trying to choose a test framework. Nightmare looks great. It provides promises, ES6, await.  

But the docs don't state if it's headless - i.e. does it open an instance of Electron for every test? (Like Selenium opens Firefox for each test).

Or is Electron just Javascript that runs in memory (like Phantom.js & Casper.js which I imagine is much faster because it doesn't have to open an entire application and Window on your screen).

Please will you add a brief note on this to your readme.md

**Comment by RichardJECooke** [comment]
Oh no, I ran an example an it opens a whole Electron window :(

So it's not headless. Please put this in the docs.  Now I can't run it in Docker to have automated tests.

**Comment by RichardJECooke** [comment]
Ahhh, now I see: 

> Pass { show: true } to the nightmare constructor to have it create a visible, rendered window that you can watch what's happening.

It would be cool if at the top of your readme you changed this line:

`var nightmare = Nightmare({ show: true })`

to 

`var nightmare = Nightmare({ show: true })  //use false for headless testing (e.g. in Docker)`

**Comment by voidmonk** [comment]
You can run it headless using xvfb or xvfb-run. See #224.

**Comment by RichardJECooke** [comment]
Are you saying if I don't have `show: true` and don't use xvfb it's still not headless?

**Comment by rosshinkley** [comment]
@RichardJECooke Whether or not you have `show: true` when running under Docker does not make it headless or headed.  It makes the rendering on the framebuffer (most likely `Xvfb`) render as though it were visible.

Nightmare isn't _truly_ headless: it requires an Electron instance to work, which in turn requires a framebuffer to render properly (at least, for now).

Setting that aside: if you set `show: true` or `show: false` (the default), it will likely not run out-of-the-box on a Docker instance as Docker instances _are_ intended to be headless, and as such, don't typically have a framebuffer included on the image.  To do that, @voidmonk's suggestion stands: have a look at #224 regarding running with `Xvfb` or `xvfb-run`.  You may also want to have  a look at the test suite for Nightmare as that runs on a headless CircleCI server.

**Comment by bchr02** [comment]
>https://github.com/segmentio/nightmare/issues/744#issuecomment-236570644

@RichardJECooke I agree, having that information there would be helpful. I just started using the example and noticed it opened a window, which is great while debugging but when I did a search on README.md for headless or hide, to my surprise I did not get an answer. So, having this would have saved me some time from continuing to dig for an answer or having to read the electron docs.
